src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -138,80 +139,99 @@ public override AssemblySymbol ContainingAssembly
 
-            var moduleSymbol = containingType.ContainingPEModule;
-            var metadataReader = moduleSymbol.Module.MetadataReader;
-            GenericParameterConstraintHandleCollection constraints;
+                PEMethodSymbol containingMethod = null;
+                PENamedTypeSymbol containingType;
 
-            try
-            {
-                constraints = metadataReader.GetGenericParameter(_handle).GetConstraints();
-            }
-            catch (BadImageFormatException)
-            {
-                constraints = default(GenericParameterConstraintHandleCollection);
-                Interlocked.CompareExchange(ref _lazyBoundsErrorInfo, new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, this), CSDiagnosticInfo.EmptyErrorInfo);
-            }
+                if (_containingSymbol.Kind == SymbolKind.Method)
+                {
+                    containingMethod = (PEMethodSymbol)_containingSymbol;
+                    containingType = (PENamedTypeSymbol)containingMethod.ContainingSymbol;
+                }
+                else
+                {
+                    containingType = (PENamedTypeSymbol)_containingSymbol;
+                }
 
-            if (constraints.Count > 0)
-            {
-                var symbolsBuilder = ArrayBuilder<TypeSymbol>.GetInstance();
-                MetadataDecoder tokenDecoder;
+                var moduleSymbol = containingType.ContainingPEModule;
+                var metadataReader = moduleSymbol.Module.MetadataReader;
+                GenericParameterConstraintHandleCollection constraints;
 
-                if ((object)containingMethod != null)
+                try
                 {
-                    tokenDecoder = new MetadataDecoder(moduleSymbol, containingMethod);
+                    constraints = metadataReader.GetGenericParameter(_handle).GetConstraints();
                 }
-                else
+                catch (BadImageFormatException)
                 {
-                    tokenDecoder = new MetadataDecoder(moduleSymbol, containingType);
+                    constraints = default(GenericParameterConstraintHandleCollection);
+                    Interlocked.CompareExchange(ref _lazyConstraintsUseSiteErrorInfo, new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, this), CSDiagnosticInfo.EmptyErrorInfo);
                 }
 
-                foreach (var constraintHandle in constraints)
+                this.GetAttributes();
+                if (_lazyHasIsUnmanagedAttribute.Value() && (_lazyFlags & (GenericParameterAttributes.ReferenceTypeConstraint | GenericParameterAttributes.DefaultConstructorConstraint)) != 0)
                 {
-                    var constraint = metadataReader.GetGenericParameterConstraint(constraintHandle);
-                    var constraintTypeHandle = constraint.Type;
+                    _lazyFlags = _lazyFlags & ~(GenericParameterAttributes.ReferenceTypeConstraint | GenericParameterAttributes.DefaultConstructorConstraint);
```

Comment:
>_lazyFlags = _lazyFlags & ~(GenericParameterAttributes.ReferenceTypeConstraint | GenericParameterAttributes.DefaultConstructorConstraint); [](start = 20, length = 138)

I don't think we should be removing any flags because there is an attribute on the type parameter.